File: current_session/python/1578.py
"""
1578
"""
class Solution:
    def minCost(self, s: str, cost: List[int]) -> int:
        prev, i = 0, 1
        c = 0
        while i < len(s):
            while i < len(s) and s[i] == s[prev]:
                i += 1
            if s[i-1] == s[prev] and i-prev > 1:
                c += sum(cost[prev:i]) - max(cost[prev:i])
            prev, i = i, i+1
        return c

File: current_session/c++/1578.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  int minCost(string s, vector<int>& cost) {
    long minCost=0;
    int prevI=0, cur=0;
    int subSum = cost[0], localMax = cost[0];
    for (int i=1;i<s.size(); ++i) {
      if (s[i] == s[prevI]) {
        subSum += cost[i];
        localMax = max(cost[i], localMax);
        cur = i;
      } else {
        if (cur != prevI) {
          minCost += (subSum - localMax);
        }
        cur = i;
        prevI = i;
        localMax = cost[i];
        subSum = cost[i];
      }
    }
    if (cur != prevI) minCost += subSum - localMax;
    return int(minCost);
  }
};